core/unix/injector.c

```diff
@@ -1169,6 +1225,22 @@ injectee_run_get_retval(dr_inject_info_t *info, void *dc, instrlist_t *ilist)
     app_pc pc;
     long r;
     ptr_int_t failure = -EUNATCH; /* Unlikely to be used by most syscalls. */
+    dr_isa_mode_t app_mode;
+
+    /* Get register state before executing the shellcode. */
+    r = our_ptrace_getregs(info->pid, &regs);
+    if (r < 0)
+        return r;
+
+#    ifdef X86
+    app_mode = IF_X64_ELSE(DR_ISA_AMD64, DR_ISA_IA32);
+#    elif defined(AARCH64)
+    app_mode = DR_ISA_ARM_A64;
+#    elif defined(ARM)
```

Comment:
nit: inconsistency between using #ifdef and defined in the same block.

Reply:
Fixed, here and in several other similar additions in this PR.